base/range.jl

```diff
@@ -116,7 +116,7 @@ _range(a::T, step::T, ::Nothing, len::Integer) where {T <: AbstractFloat} =
 _range(a::T, step, ::Nothing, len::Integer) where {T} =
     _rangestyle(OrderStyle(T), ArithmeticStyle(T), a, step, len)
 _rangestyle(::Ordered, ::ArithmeticWraps, a::T, step::S, len::Integer) where {T,S} =
-    StepRange{T,S}(a, step, convert(T, a+step*(len-1)))
+    StepRange{typeof(a+0*step),S}(a, step, a+step*(len-1))
```

Comment:
I think `promote_type(T, S)` is better here.

Reply:
That won’t work correctly for ranges of `Char`s (`promote_type(Char, Int) == Any`). However, `a+0*step` should probably be replaced by `a+false*step` to avoid unnecessary widening.